opentelemetry-dynatrace/src/transform/metrics.rs

```diff
@@ -0,0 +1,1165 @@
+}
+
+impl MetricLine {
+    /// Create a new `MetricLine`.
+    pub fn new(key: MetricKey, kind: NumberKind) -> Self {
+        MetricLine {
+            key,
+            kind,
+            dimensions: None,
+            min: None,
+            max: None,
+            sum: None,
+            count: None,
+            delta: None,
+            gauge: None,
+            timestamp: None,
+        }
+    }
+
+    /// Common attributes that apply to this metric line.
+    pub fn dimensions(mut self, dimensions: Option<DimensionSet>) -> Self {
+        self.dimensions = dimensions;
+        self
+    }
+
+    /// The min value.
+    pub fn min(mut self, min: Option<Number>) -> Self {
+        self.min = min;
+        self
+    }
+
+    /// The max value.
+    pub fn max(mut self, max: Option<Number>) -> Self {
+        self.max = max;
+        self
+    }
+
+    /// The sum value.
+    pub fn sum(mut self, sum: Option<Number>) -> Self {
+        self.sum = sum;
+        self
+    }
+
+    /// The count value.
+    pub fn count(mut self, count: Option<u64>) -> Self {
+        self.count = count;
+        self
+    }
+
+    /// The delta value.
+    pub fn delta(mut self, delta: Option<Number>) -> Self {
+        self.delta = delta;
+        self
+    }
+
+    /// The gauge value.
+    pub fn gauge(mut self, gauge: Option<Number>) -> Self {
+        self.gauge = gauge;
+        self
+    }
+
+    /// The timestamp in UTC milliseconds.
+    /// Allowed range is between 1 hour into the past and 10 minutes into the future from now.
+    /// If no timestamp is provided, the ingestion time of the Dynatrace server will be used automatically.
+    pub fn timestamp(mut self, timestamp: Option<u64>) -> Self {
+        self.timestamp = timestamp;
+        self
+    }
+}
+
+impl fmt::Display for MetricLine {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        fmt.write_str(&format!("{}", &self.key))?;
+        fmt.write_char(',')?;
+
+        if let Some(dimensions) = self.dimensions.to_owned() {
+            fmt.write_str(&format!("{}", dimensions))?;
+        }
+
+        if self.min.is_some() && self.max.is_some() && self.sum.is_some() && self.count.is_some() {
+            fmt.write_char(' ')?;
+            fmt.write_fmt(format_args!(
+                "gauge,min={},max={},sum={},count={}",
+                convert(&self.kind, self.min.to_owned().unwrap()),
+                convert(&self.kind, self.max.to_owned().unwrap()),
+                self.sum.to_owned().unwrap().to_f64(&self.kind),
+                self.count.to_owned().unwrap(),
+            ))?;
+        } else if let Some(delta) = self.delta.to_owned() {
+            fmt.write_char(' ')?;
+            fmt.write_fmt(format_args!("count,delta={}", convert(&self.kind, delta)))?;
+        } else if let Some(gauge) = self.gauge.to_owned() {
+            fmt.write_char(' ')?;
+            fmt.write_fmt(format_args!("gauge,{}", convert(&self.kind, gauge)))?;
+        }
```

Comment:
I think we can simplify something like `gauge,min=80.6,max=80.6,sum=80.6,count=1` to `80.6`. Maybe it will be helpful to add it here also.

Reply:
Yes, very good idea. I implemented this change and updated the integration test.